core/src/main/java/org/elasticsearch/index/translog/BufferingTranslogWriter.java

```diff
@@ -107,15 +119,24 @@ public void sync() throws IOException {
             return;
         }
         synchronized (this) {
+            ensureOpen();
             channelReference.incRef();
             try {
+                final long offsetToSync;
                 try (ReleasableLock lock = writeLock.acquire()) {
                     flush();
-                    lastSyncedOffset = totalOffset;
+                    offsetToSync = totalOffset;
                 }
                 // we can do this outside of the write lock but we have to protect from
                 // concurrent syncs
-                checkpoint(lastSyncedOffset, operationCounter, channelReference);
+                try {
+                    ensureOpen();
+                    checkpoint(offsetToSync, operationCounter, channelReference);
```

Comment:
should we move this to the write lock? I'm worried now we will write  an offsetToSync that is not correlated to the operation counter.